api/hypershift/v1beta1/certificaterevocationrequest_types.go

```diff
@@ -0,0 +1,96 @@
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	SchemeBuilder.Register(&CertificateRevocationRequest{})
+	SchemeBuilder.Register(&CertificateRevocationRequestList{})
+}
+
+// +genclient
+// +kubebuilder:resource:path=certificaterevocationrequest,shortName=crr;crrs,scope=Namespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+
+// CertificateRevocationRequest defines the desired state of CertificateRevocationRequest
+type CertificateRevocationRequest struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   CertificateRevocationRequestSpec   `json:"spec,omitempty"`
+	Status CertificateRevocationRequestStatus `json:"status,omitempty"`
+}
+
+// CertificateRevocationRequestSpec defines the desired state of CertificateRevocationRequest
+type CertificateRevocationRequestSpec struct {
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=customer-break-glass
```

Comment:
Use golang enum.  Enumerated values use CamelCase in our API.

Reply:
Is there an example of what you mean? Golang doesn't have enums ...

Reply:
`CustomerBreakGlass`

Reply:
Oops - TODO